elyra/pipeline/component.py

```diff
@@ -360,13 +341,9 @@ def determine_type_information(self, type: str):
             default_value = False
         elif type_lowered in ['dict', 'dictionary']:
             type_lowered = "dictionary"
-            default_value = ''
         elif type_lowered in ['list', 'set', 'array', 'arr']:
             type_lowered = "list"
-            default_value = ''
-        elif type_lowered in ['file']:
-            default_value = ''
-        else:
+        elif type_lowered not in ['file']:
```

Comment:
This is correct, but it requires the reader to think about the `not` condition and can be done simply because there's only one entry in the 'file' list.  Should another type mapping be introduced or we find multiple mappings for 'file' someday, then we'd need to abandon this approach.  I think this might be easier to read and maintain at the expense of some redundancy:
```python
    elif type_lowered in ['file']:
        type_lowered = 'file'
    else:
        type_lowered = 'string'
```

Reply:
I had this thought as well! Will change back to a more readable format